lib/resources/datasets.js

```diff
@@ -26,6 +27,7 @@ module.exports = (service, endpoint) => {
     Datasets.get(params.projectId, params.name)
       .then(getOrNotFound)
       .then((dataset) => auth.canOrReject('dataset.read', dataset)
+        .then(rejectIf((() => dataset.publishedAt == null), noargs(Problem.user.notFound)))
```

Comment:
Rather than adding a separate check, I think you could use the optional third parameter of `Datasets.get()`. That's how some other endpoints are returning a 404 (for example, lib/resources/entities.js).